iib/workers/tasks/utils.py

```diff
@@ -163,11 +172,21 @@ def skopeo_inspect(*args, return_json=True):
             exc_msg = f'Failed to inspect {arg}. Make sure it exists and is accessible to IIB.'
             break
 
+    cache_key = generate_cache_key(skopeo_inspect.__name__, *args, return_json=return_json)
     skopeo_timeout = get_worker_config().iib_skopeo_timeout
     cmd = ['skopeo', '--command-timeout', skopeo_timeout, 'inspect'] + list(args)
+
+    if use_cache:
+        # get data from cache
+        output_cache = dogpile_cache.get(cache_key)
+        if output_cache:
+            return output_cache
+
     output = run_cmd(cmd, exc_msg=exc_msg)
     if return_json:
-        return json.loads(output)
+        output = json.loads(output)
+
+    dogpile_cache.set(cache_key, output)
```

Comment:
Is there a JSON/non-JSON control implemented? It seems like the cache is set to whatever type was last used. Doesn't that mean that the next call might return the wrong type?

Reply:
The key is generated based on arguments (arg, kwarg) You can check [generate_cache_key](https://github.com/release-engineering/iib/pull/142/files#diff-e93fe72aa218e5b071f12554fcb09d85R11) and it is called on line 175: `cache_key = generate_cache_key(skopeo_inspect.__name__, *args, return_json=return_json)` 
So the key is generated different for both values (True/False)